pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -149,11 +151,25 @@ func (p *ConfigurationEditFlags) addSharedFlags(command *cobra.Command) {
 	// Don't mark as changing the revision.
 
 	command.Flags().StringArrayVarP(&p.Annotations, "annotation", "a", []string{},
-		"Service annotation to set. name=value; you may provide this flag "+
+		"Annotations to set for both Service and Revision. name=value; you may provide this flag "+
 			"any number of times to set multiple annotations. "+
 			"To unset, specify the annotation name followed by a \"-\" (e.g., name-).")
 	p.markFlagMakesRevision("annotation")
 
+	command.Flags().StringArrayVarP(&p.AnnotationsService, "annotation-service", "", []string{},
+		"Service annotation to set. name=value; you may provide this flag "+
+			"any number of times to set multiple annotations. "+
+			"To unset, specify the annotation name followed by a \"-\" (e.g., name-). This flag takes "+
+			"precedence over the \"annotation\" flag.")
+	p.markFlagMakesRevision("annotation-service")
+
+	command.Flags().StringArrayVarP(&p.AnnotationsRevision, "annotation-revision", "", []string{},
+		"Revision annotation to set. name=value; you may provide this flag "+
+			"any number of times to set multiple annotations. "+
+			"To unset, specify the annotation name followed by a \"-\" (e.g., name-). This flag takes "+
+			"precedence over the \"annotation\" flag.")
```

Comment:
I can comment out lines 171 and/or 164 and all tests pass!

Reply:
Have refactored the tests